### PR TITLE
Move Integration Tests to dedicated project

### PR DESCRIPTION
**Pull request by coquagli**
Add changes for [514](https://github.com/Microsoft/vscode-mssql/issues/514). Add IntegrationTests project. Move all tests ifdef'd with LIVE_CONNECTION_TESTS to IntegrationTests project in identical locations. Delete files that have no remaining code. 

If I am missing any changes that are needed to enable to run this on Jenkins, please let me know.

### CODE REVIEW

File: sqltoolsservice.sln [hunk 1]
```diff
@@ -53,28 +53,45 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "CodeCoverage", "CodeCoverag
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.SqlTools.ServiceLayer.PerfTests", "test\Microsoft.SqlTools.ServiceLayer.PerfTests\Microsoft.SqlTools.ServiceLayer.PerfTests.xproj", "{7E5968AB-83D7-4738-85A2-416A50F13D2F}"
 EndProject
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.SqlTools.ServiceLayer.IntegrationTests", "test\Microsoft.SqlTools.ServiceLayer.IntegrationTests\Microsoft.SqlTools.ServiceLayer.IntegrationTests.xproj", "{08AF0209-D598-47BB-9DFD-FC9E74C0FE56}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
+		Integration|Any CPU = Integration|Any CPU
```
Comment (coquagli) [inline sqltoolsservice.sln]: These "Integration" GUIDs got automatically added. Not sure what they do?

Comment (benrr101) [inline sqltoolsservice.sln]: This is pretty much the expected behavior. It just sets up which projects should build and what configuration they should use when building the solution.

Comment (coquagli) [inline sqltoolsservice.sln]: Ahh, okay.

**Comment by kburtram** [comment]
It looks like the coveralls coverage number is dropping by 15% with this change.  Could you please resolve this issue before merging this change.

**Comment by kevcunnane** [comment]
To fix coveralls, you'll need to edit the AppVeyor build to run your new Test DLL in the coverage run. You should be able to log in with your GitHub account and make the necessary changes.

**Comment by coquagli** [comment]
Do I need to change [appveyor.yml](https://github.com/Microsoft/sqltoolsservice/blob/9140f02b0d6f6e5600bbc16353882de22c1e54c4/appveyor.yml)? This is the only related file I could find, but the comments in the file suggest that I'm not looking in the correct place.

**Comment by kevcunnane** [comment]
You need to update codecoverage.bat under test\CodeCoverage:

Add build for your test project to following line:
```
REM run the tests through OpenCover and generate a report
dotnet build %WORKINGDIR%..\..\test\Microsoft.SqlTools.ServiceLayer.Test\project.json %DOTNETCONFIG% 
dotnet build %WORKINGDIR%..\..\test\Microsoft.SqlTools.ServiceLayer.TestDriver\project.json %DOTNETCONFIG%
```

Execute test by copying one of the following lines but using your project:

```

"%WORKINGDIR%packages\OpenCover.4.6.519\tools\OpenCover.Console.exe" -mergeoutput -register:user -target:dotnet.exe -targetargs:"test %WORKINGDIR%..\Microsoft.SqlTools.ServiceLayer.TestDriver\project.json %DOTNETCONFIG%" -oldstyle -filter:"+[Microsoft.SqlTools.*]* -[xunit*]*" -output:coverage.xml -searchdirs:%WORKINGDIR%..\Microsoft.SqlTools.ServiceLayer.TestDriver\bin\Debug\netcoreapp1.0

"%WORKINGDIR%packages\OpenCover.4.6.519\tools\OpenCover.Console.exe" -mergeoutput -register:user -target:dotnet.exe -targetargs:"test %WORKINGDIR%..\Microsoft.SqlTools.ServiceLayer.Test\project.json %DOTNETCONFIG%" -oldstyle -filter:"+[Microsoft.SqlTools.*]* -[xunit*]*" -output:coverage.xml -searchdirs:%WORKINGDIR%..\Microsoft.SqlTools.ServiceLayer.Test\bin\Debug\netcoreapp1.0

```

If you add this into your PR, it should work when running code coverage, and you'll get verification by seeing the coverage numbers staying the same.

**Comment by coquagli** [comment]
Got it. Thanks, Kevin!

**Pull request merged**